### DISCUSSION TITLE
Continuation of issue #11868: target by state ID

### DISCUSSION THREAD

**Discussion by Mrten** [saltstack/salt#67504] [Feature Requests]
I'd like to target highstate runs by state. This way I could, say, restart all apaches at the same time and be sure that I haven't missed any.

Like this:

salt -S apache state.highstate

Explanation: I write my states for, say, apache, with one basic state that sets everything that I want the same for all apaches (like the LogFormat, the mimesettings, etc) and include: that basic state from every other apache that gets deployed, however different its usecase may be.

When one is then confronted with something like the Heartbleed OpenSSL bug one would like to upgrade/restart all those apaches in one go.

I could work around this with pillar values but that feels like a kludge to me.

Workarounds with jq exist, see [here](https://github.com/saltstack/salt/issues/11868#issuecomment-705208541).

**Comment by Mrten** [comment]
There's lots of input from other folks on the referred issue.

**Comment by dmurphy18** [comment]
@Mrten One of the reason for closing out old issues, circa 10 years ago, was to use the new template with issues etc. stating, bug, feature request, etc. which allows for better tracking etc.

Can you please update this accordingly,since this is a Feature Request,  if you want the issue to be considered.  Noting you had Feature Request on the old issue from 2014.
